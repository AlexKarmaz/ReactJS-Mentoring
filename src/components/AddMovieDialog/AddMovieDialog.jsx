import React, {useCallback} from 'react';
import ModalDialog from '../Common/ModalDialog';
import MovieForm from '../Common/MovieForm'

const AddMovieDialog = ({onClose}) => {
    const onDialogClose = useCallback(() => onClose(), [onClose]);

    return (
        <ModalDialog dialogTitle='Add movie' onDialogClose={onDialogClose}>
            <MovieForm onReset={onDialogClose} onSubmit={() => console.log('submit')}/>
        </ModalDialog>
    );
};

export default AddMovieDialog
