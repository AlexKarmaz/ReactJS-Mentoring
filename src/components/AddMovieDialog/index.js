import AddMovieDialog from './AddMovieDialog';

export default AddMovieDialog;
