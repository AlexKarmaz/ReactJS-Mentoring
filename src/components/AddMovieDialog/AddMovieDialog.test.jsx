import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import AddMovieDialog from './AddMovieDialog';

describe('AddMovieDialog', () => {
  test('should be rendered correctly', () => {
    const initialState = {
        moviesData: {
          genres: ['Comedy', 'Drama', 'Horror'],
        }
    };
    const store = configureStore([])(initialState);

    const form = mount(
      <Provider store={store}>
        <AddMovieDialog />
      </Provider>,
    );

    expect(form.find('.modalDialog-title').text()).toEqual('Add movie');
    expect(form).toMatchSnapshot();
  });
});
