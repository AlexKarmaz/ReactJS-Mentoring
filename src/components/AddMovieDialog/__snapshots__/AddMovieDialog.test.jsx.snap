// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieDialog should be rendered correctly 1`] = `ReactWrapper {}`;
